extensions/authn/filter_context.h

```diff
@@ -0,0 +1,97 @@
+/* Copyright 2020 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/config/filter/http/authn/v2alpha2/config.pb.h"
+#include "envoy/network/connection.h"
+#include "extensions/filters/http/well_known_names.h"
+#include "src/istio/authn/context.pb.h"
+
+namespace Extensions {
+namespace AuthN {
+
+using Envoy::Http::RequestHeaderMap;
+using Envoy::Network::Connection;
+
+// FilterContext holds inputs, such as request dynamic metadata and connection
+// and result data for authentication process.
+class FilterContext {
+ public:
+  FilterContext(
+      const envoy::config::core::v3::Metadata& dynamic_metadata,
+      const RequestHeaderMap& header_map, const Connection* connection,
```

Comment:
This seems to be using Envoy interfaces. How are you going to compile this to Wasm?

Reply:
I'm not considering about wasm compilation in this PR. I will submit a PR which will include plugin.h/.cc using proxy-wasm interface. I will fix them in that.

Reply:
I'm just wondering that if we're revising the implementation anyways, why not write it once with Wasm ABI.

Reply:
This PR doesn't include the build components which will be required to build AuthN filter. I'd like to split AuthN wasm implementations described in #2800. If we consider about wasm build in this PR. It will be complex to review all of codes, and out of request authenticator context. We have had already the implementation to be available not NullVM in #2800.